src/content/protocols/compound-v2.md

```diff
@@ -0,0 +1,227 @@
+---
+protocol: "Compound V2"
+website: "https://compound.finance/"
+x: "https://x.com/compoundfinance"
+github: ["https://github.com/compound-finance"]
+defillama_slug: ["compound-v2"]
+chain: "Ethereum"
+stage: 0
+reasons: []
+risks: ["L", "H", "H", "H", "L"]
+author: ["eliasbourgon"]
+submission_date: "2025-05-13"
+publish_date: "1970-01-01"
+update_date: "1970-01-01"
+---
+
+# Summary
+
+Compound v2 is a lending protocol where each market accepts one “base” asset (e.g. USDC, DAI, ETH via cETH, WBTC, BAT, REP, ZRX) as liquidity. Users can deposit that base asset to earn interest, or borrow it with a variety of other assets as collateral. Every base asset market runs in its own isolated instance, with its own price oracle, interest-rate model, reserves and risk parameters.
+
+# Ratings
+
+## Chain
+
+Compound-v2 is deployed on various chains. This review is based on the Ethereum mainnet deployment.
+
+> Chain score: Low
+
+## Upgradeability
+
+The Compound-v2 protocol is almost fully upgradeable allowing for the update of governance and markets logic, with implementation contracts such as `CErc20Delegate`, `Comptroller` and `GovernorBravoDelegate` . This can result in the _loss of funds or unclaimed yield_ as well as lead to other changes in the expected performance of the protocol.
+
+The protocol's governance, through the Timelock contract, controls key protocol parameters, including interest rate models per asset. Most markets operate through upgradeable proxies pointing to `CToken` implementations. The `Timelock`, controlled by the governance, can update the markets' implementation, modify interest rates, and withdraw accumulated reserves. This means that a sufficiently powerful proposal could install a rate model that sets borrowing costs to extremes or redirect reserves out of the protocol. There are a few notable exceptions with unupgradeable markets using `CErc20` or `CEther` contracts.
+
+Compound v2 currently relies on a non-upgradeable oracle contract to deliver price feeds via Chainlink. While the contract itself cannot be upgraded, its configuration is governed by the Compound community multisig with broad administrative powers: it can change the Chainlink price feed source, add or remove asset markets, or set a fixed fallback price for individual tokens. This power on the price feed could be abused and lead to _loss of user funds_ through unlunjustified liquidations.
```

Comment:
```suggestion
Compound v2 currently relies on a non-upgradeable oracle contract to deliver price feeds via Chainlink. While the contract itself cannot be upgraded, its configuration is governed by the Compound community multisig with broad administrative powers: it can change the Chainlink price feed source, add or remove asset markets, or set a fixed fallback price for individual tokens. This power on the price feed could be abused and lead to _loss of user funds_ through unjustified liquidations.
```